src/SayMoreTests/model/SessionArchivingTests.cs

```diff
@@ -320,51 +322,223 @@ public void GetOneLanguage_UndeterminedIso_ReturnsNull()
+					   where f.TimePosition.End != null
+					   where f.TimePosition.End != "Unspecified"
+					   select f.TimePosition.End).FirstOrDefault();
+			Assert.AreEqual(sampleDuration, val);
+			AddIMDISessionTestVerify(model, imdiSession, sourceMediaFile1);
+		}
+
+		[Test]
+		public void AddIMDISession_PersonCode_AddActorHasSameCode()
+		{
+			const string sampleCode = "123";
+			_personMetaFile.Setup(m => m.GetStringValue(PersonFileType.kCode, It.IsAny<string>())).Returns(sampleCode);
+			var model = AddIMDISessionTestSetup(out var imdiSession);
+			ArchivingHelper.AddIMDISession(_session, model.Object);
+			var val = (from a in imdiSession.Object.MDGroup.Actors.Actor
+				where !string.IsNullOrEmpty(a.Code)
+				select a.Code).FirstOrDefault();
+			Assert.AreEqual(sampleCode, val);
+			AddIMDISessionTestVerify(model, imdiSession);
+		}
+
+		private Mock<IMDIArchivingDlgViewModel> AddIMDISessionTestSetup(out Mock<SIL.Archiving.IMDI.Schema.Session> imdiSession, ComponentFile mediaFile = null)
 		{
 			var model = new Mock<IMDIArchivingDlgViewModel>(MockBehavior.Strict, "SayMore", "ddo", "ddo-session", "whatever",
 				false, null, @"C:\my_imdi_folder");
-			var imdiSession = new Mock<SIL.Archiving.IMDI.Schema.Session>(MockBehavior.Strict);
+			imdiSession = new Mock<SIL.Archiving.IMDI.Schema.Session>(MockBehavior.Strict);
 			imdiSession.Object.Name = "ddo";
 			model.Setup(m => m.AddSession(_session.Id)).Returns(imdiSession.Object);
-			_session._mediaFiles = new ComponentFile[0];
-			const string SampleEthnicGroup = "Ewondo";
-			_personMetaFile.Setup(m => m.GetStringValue("ethnicGroup", It.IsAny<string>())).Returns(SampleEthnicGroup);
-			ArchivingHelper.AddIMDISession(_session, model.Object);
-			var actor = imdiSession.Object.MDGroup.Actors.Actor.FirstOrDefault();
-			Assert.AreEqual(SampleEthnicGroup, actor.EthnicGroup);
+			_session._mediaFiles = mediaFile == null ? new ComponentFile[0] : new[] { mediaFile };
+			return model;
+		}
+
+		private Mock<ComponentFile> IMDIMediaFileSetup()
+		{
+			var fileType = new Mock<AudioFileType>(null, null, null);
+			fileType.Setup(t => t.IsAudioOrVideo).Returns(true);
+			var sourceMediaFile1 = new Mock<ComponentFile>(MockBehavior.Strict, fileType.Object);
+			sourceMediaFile1.Setup(f => f.PathToAnnotatedFile).Returns(_mp3FullName);
+			return sourceMediaFile1;
+		}
+
+		private static void AddIMDISessionTestVerify(Mock<IMDIArchivingDlgViewModel> model, Mock<SIL.Archiving.IMDI.Schema.Session> imdiSession, Mock<ComponentFile> sourceMediaFile1 = null)
+		{
+			model.VerifyAll();
+			imdiSession.VerifyAll();
+			sourceMediaFile1?.VerifyAll();
 		}
 		#endregion
 	}
 
 	public class DummySession : Session
 	{
 		public string[] _participants;
-		private readonly Mock<ProjectElementComponentFile> _metaFile = new Mock<ProjectElementComponentFile>();
+		public readonly Mock<ProjectElementComponentFile> _metaFile = new Mock<ProjectElementComponentFile>();
```

Comment:
_ in front of a variable indicates that it is private. So this should be metaFile or MetaFile